optuna/storages/_grpc/client.py

```diff
@@ -67,28 +73,35 @@ class GrpcStorageProxy(BaseStorage):
     """
 
     def __init__(self, *, host: str = "localhost", port: int = 13000) -> None:
-        self._stub = api_pb2_grpc.StorageServiceStub(
-            grpc.insecure_channel(
-                f"{host}:{port}",
-                options=[("grpc.max_receive_message_length", -1)],
-            )
-        )  # type: ignore
-        self._cache = GrpcClientCache(self._stub)
         self._host = host
         self._port = port
+        self.setup()
+
+    def setup(self) -> None:
```

Comment:
[nit] How about making this method private? Unlike other methods such as `wait_server_ready` and `close` methods, I have no idea how users can utilize this method for now.

```suggestion
    def _setup(self) -> None:
```